trulens_eval/examples/expositional/end2end_apps/streamlit_app/app.py

```diff
@@ -43,4 +44,7 @@ def generate_response(input_text):
 
 if submitted:
     trulens_st.trulens_feedback(record=record)
-    trulens_st.trulens_trace(record=record)
+    app = tru.get_app(app_id=record.app_id)
+    records, feedback = tru.get_records_and_feedback()
+    record_json = json.loads(records.loc[records['record_id'] == record.record_id]['record_json'].values[0])
+    trulens_st.trulens_trace(record_json=record_json, app=app)
```

Comment:
is there a reason you changed to this call pattern over the simpler `trulens_st.trulens_trace(record=record)` you had before?

Reply:
Yes - before I was calling Tru internally which was not ideal since the user may have different Tru specification. Better if they get the data themselves from their Tru connection.

Reply:
In that case, shouldn't the `trulens_leaderboard` also do that?

Also it may be easier for the user to just take in the `tru` and the `record` since what's going on in L49 is a bit complex for the user I'd say.